Gems/ScriptAutomation/Code/Source/Utils/Utils.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <Utils/Utils.h>
+
+#include <AtomCore/Instance/InstanceDatabase.h>
+#include <Atom/RHI/RHISystemInterface.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/RPISystemInterface.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/Image/ImageSystemInterface.h>
+#include <Atom/RPI.Reflect/Asset/AssetUtils.h>
+#include <Atom/RPI.Reflect/Image/StreamingImageAssetCreator.h>
+#include <Atom/RPI.Reflect/Image/ImageMipChainAssetCreator.h>
+#include <Atom/Utils/DdsFile.h>
+
+#include <AzFramework/CommandLine/CommandLine.h>
+#include <AzFramework/API/ApplicationAPI.h>
+#include <AzFramework/Windowing/NativeWindow.h>
+#include <AzCore/Asset/AssetCommon.h>
+#include <AzCore/PlatformIncl.h>
+#include <AzCore/IO/SystemFile.h>
+#include <AzCore/IO/Path/Path.h>
+
+
+namespace ScriptAutomation
+{
+    namespace Utils
+    {
+        void ResizeClientArea(uint32_t width, uint32_t height)
+        {
+            AzFramework::NativeWindowHandle windowHandle = nullptr;
+            AzFramework::WindowSystemRequestBus::BroadcastResult(
+                windowHandle,
+                &AzFramework::WindowSystemRequestBus::Events::GetDefaultWindowHandle);
+
+            AzFramework::WindowSize clientAreaSize = {width, height};
+            AzFramework::WindowRequestBus::Event(windowHandle, &AzFramework::WindowRequestBus::Events::ResizeClientArea, clientAreaSize);
+            AzFramework::WindowSize newWindowSize;
+            AzFramework::WindowRequestBus::EventResult(newWindowSize, windowHandle, &AzFramework::WindowRequests::GetClientAreaSize);
+            AZ_Error("ResizeClientArea", newWindowSize.m_width == width && newWindowSize.m_height == height,
+                "Requested window resize to %ux%u but got %ux%u. This display resolution is too low or desktop scaling is too high.",
+                width, height, newWindowSize.m_width, newWindowSize.m_height);
+        }
+
+        bool SupportsToggleFullScreenOfDefaultWindow()
+        {
+            return AzFramework::NativeWindow::CanToggleFullScreenStateOfDefaultWindow();
+        }
+
+        void ToggleFullScreenOfDefaultWindow()
+        {
+            AzFramework::NativeWindow::ToggleFullScreenStateOfDefaultWindow();
+        }
+
+        AZStd::string GetScreenshotsPath(bool resolvePath)
+        {
+            AZStd::string path = "@user@/scriptautomation/screenshots/";
+
+            if (resolvePath)
+            {
+                path = Utils::ResolvePath(path);
+            }
+
+            return path;
+        }
```

Comment:
This can return an AZ::IO::FixedMaxPath or an AZ::IO::Path.

```suggestion
        AZ::IO::FixedMaxPath GetScreenshotsPath(bool resolvePath)
        {
            AZ::IO::FixedMaxPath path("@user@");
            if (resolvePath)
            {
                if (auto settingsRegistry = AZ::SettingsRegistry(); settingsRegistry != nullptr))
                {
                    resolvePath.clear();
                    settingsRegistry->Get(path.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectUserPath
                }
            }
            
            path /= "scriptautomation/screenshots";
            
            return path.LexicallyNormal();
```